xbmc/platform/android/activity/JNIXBMCDisplayManagerDisplayListener.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (C) 2016 Christian Browet
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "JNIXBMCDisplayManagerDisplayListener.h"
+#include <androidjni/jutils-details.hpp>
+
+#include <androidjni/Context.h>
+#include "CompileInfo.h"
+
+#include "XBMCApp.h"
+
+using namespace jni;
+
+static std::string s_className = std::string(CCompileInfo::GetClass()) + "/interfaces/XBMCDisplayManagerDisplayListener";
+
+CJNIXBMCDisplayManagerDisplayListener::CJNIXBMCDisplayManagerDisplayListener()
+  : CJNIBase(s_className)
+{
+  m_object = new_object(CJNIContext::getClassLoader().loadClass(GetDotClassName(s_className)));
+  m_object.setGlobal();
+}
+
+void CJNIXBMCDisplayManagerDisplayListener::RegisterNatives(JNIEnv* env)
+{
+  jclass cClass = env->FindClass(s_className.c_str());
+  if(cClass)
+  {
+    JNINativeMethod methods[] =
+    {
+      {"_onDisplayAdded", "(I)V", (void*)&CJNIXBMCDisplayManagerDisplayListener::_onDisplayAdded},
```

Comment:
C++ style cast

Reply:
no, its a function definition

Reply:
`&CJNIXBMCDisplayManagerDisplayListener::_onDisplayAdded` is a function pointer with type `void (*)(JNIEnv *, jobject, int)`  and it is casted to `void*`. So correct is `reinterpret_cast<void*>(&CJNIXBMCDisplayManagerDisplayListener::_onDisplayAdded)`